Bugsnag/KSCrash/Source/KSCrash/Recording/Sentry/BSG_KSCrashSentry.c

```diff
@@ -229,7 +229,7 @@ bool bsg_kscrashsentry_beginHandlingCrash(const thread_t offender) {
         return true;
     }
 
-    if (offender == firstHandlingThread) {
+    if (bsg_g_context->handlingCrash) {
```

Comment:
`if (offender == firstHandlingThread)` positively checks that the crash handler crashed, rather than just a concurrent crash outside the crash handler.

Reply:
`offender == firstHandlingThread` is what leads to the false "crash in crash reporter" message, because the thread that captures the C++ or NSException is the offending thread. So when it goes through this path twice (once for C++, once for NSException), it triggers.